File: py/soal3.py
# ANGGOTA KELOMPOK
# - Naufal Fahrezi (14081020035)
# - Amalia Nur Fitri (140810200045)
# - Calvin Calfi Montolalu (140810200053)
# - Deani Puteri Virdiana (140810200059)

UPAH=50

class Pegawai :
    def __init__(self, kode, nama, masuk, keluar) :
        self.__kode=kode
        self.__nama=nama
        self.__jamMasuk=masuk
        self.__jamKeluar=keluar

    def setKode(self,kode) : self.__kode=kode
    def setNama(self,nama) : self.__nama=nama
    def setJamMasuk(self,masuk) : self.__jamMasuk=masuk
    def setJamKeluar(self,keluar) : self.__jamKeluar=keluar

    def getKode(self) : return self.__kode
    def getNama(self) : return self.__nama
    def getJamMasuk(self) : return self.__jamMasuk
    def getJamKeluar(self) : return self.__jamKeluar

    def hitungJamKerja(self) :
        tempMasuk=8.0 if self.__jamMasuk < 8.0 else self.__jamMasuk
        tempKeluar=16.0 if self.__jamKeluar > 16.0 else self.__jamKeluar
        return tempKeluar-tempMasuk

    def hitungUpah(self) : return self.hitungJamKerja() * UPAH

    def print(self) :
        print("Kode Pegawai = " + self.__kode)
        print("Nama Pegawai = " + self.__nama)
        print("Jam Masuk = " + self.__jamMasuk)
        print("Jam Keluar = " + self.__jamKeluar)
        print("Lama Kerja = " + self.__hitungJamKerja)
        print("Total Upah = " + self.__hitungUpah)

File: cpp/soal3.cpp
#include <iostream>
#include <string>
#include "waktu.cpp"

using namespace std;

class Pegawai {
    private:
        int kode;
        string nama;
        double jamMasuk;
        double jamKeluar;
        const int UPAH=50;

    public:
        Pegawai() {
            this->kode=0;
            this->nama="";
            this->jamMasuk=0.0;
            this->jamKeluar=0.0;
        }

        Pegawai(int kode, string nama, double masuk, double keluar) {
            this->kode=kode;
            this->nama=nama;
            this->jamMasuk=masuk;
            this->jamKeluar=keluar;
        }

        void setKode(int kode) { this->kode=kode; }
        void setNama(string nama) { this->nama=nama; }
        void setJamMasuk(double masuk) { this->jamMasuk=masuk; }
        void setJamKeluar(double keluar) { this->jamKeluar=keluar; }

        int getKode() { return this->kode; }
        string getNama() { return this->nama; }
        double getJamMasuk() { return this->jamMasuk; }
        double getJamKeluar() { return this->jamKeluar; }


        void input() {
            cout << "Masukkan kode pegawai: "; cin >> this->kode;
            cout << "Masukkan nama pegawai: "; cin >> this->nama;
            cout << "Masukkan jam masuk: "; cin >> this->jamMasuk;
            cout << "Masukkan jam keluar: "; cin >> this->jamKeluar;
        }

        void print() {
            cout << this->kode << "\t\t"
                 << this->nama << "\t\t\t"
                 << this->jamMasuk << "\t\t\t"
                 << this->jamKeluar << "\t\t\t"
                 << this->hitungJamKerja() << "\t\t\t"
                 << this->hitungUpah() << "\t\t\t";
        }

        double hitungJamKerja() {
            double tempMasuk=(this->jamMasuk < 8.0) ? 8.0 : this->jamMasuk;
            double tempKeluar=(this->jamKeluar > 16.0) ? 16.0 : this->jamKeluar;
            return tempKeluar-tempMasuk;
        }

        double hitungUpah() {
            return this->hitungJamKerja() * UPAH;
        }
};

int main() {
    Pegawai P1(200035, "Naufal", 7.5, 17.2);
    Pegawai P2;
    Pegawai P3;
    Pegawai P4(200059, "Deani", 9.5, 19.7);

    P2.setKode(200045);
    P2.setNama("Amalia");
    P2.setJamMasuk(8.5);
    P2.setJamKeluar(14.5);

    P3.input();
    cout << "==================================================================================\n";
    cout << "| Kode Pegawai | \tNama\t | Waktu Datang | Waktu Pulang | Waktu Kerja | \tUpah\t |\n";
    cout << "==================================================================================\n";
    P1.print();
    P2.print();
    P3.print();
    cout << P4.getKode() << "\t\t"
         << P4.getNama() << "\t\t\t"
         << P4.getJamMasuk() << "\t\t\t"
         << P4.getJamKeluar() << "\t\t\t"
         << P4.hitungJamKerja() << "\t\t\t"
         << P4.hitungUpah() << "\t\t\t";
}